2023/08/wtchangdm/day8.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy)]
+enum Direction {
+    Left,
+    Right,
+}
+
+impl Direction {
+    fn from(ch: &char) -> Self {
+        match ch {
+            'L' => Self::Left,
+            'R' => Self::Right,
+            _ => panic!("come on it's impossible"),
+        }
+    }
+}
+
+struct Map {
+    cursor: usize,
+    directions: Vec<Direction>,
+    nodes: HashMap<String, [String; 2]>,
+}
+
+impl Map {
+    fn from(input: &[String]) -> Self {
```

Comment:
```suggestion
    fn from(input: &[&str]) -> Self {
```

We end up calling `to_string` or `into` to clone the String, so could we just make the type `&[&str]` so no double clones needed?

And the `input` in tests could be mininized to 

```rust
        let input = [
            "RL",
            "",
            "AAA = (BBB, CCC)",
            "BBB = (DDD, EEE)",
            "CCC = (ZZZ, GGG)",
            "DDD = (DDD, DDD)",
            "EEE = (EEE, EEE)",
            "GGG = (GGG, GGG)",
            "ZZZ = (ZZZ, ZZZ)",
        ];
```